src/coreclr/src/vm/processdiagnosticsprotocolhelper.cpp

```diff
@@ -0,0 +1,210 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "common.h"
+#include "fastserializer.h"
+#include "processdiagnosticsprotocolhelper.h"
+#include "eventpipeeventsource.h"
+#include "diagnosticsprotocol.h"
+
+#ifdef FEATURE_PERFTRACING
+
+static bool IsNullOrWhiteSpace(LPCWSTR value)
+{
+    if (value == nullptr)
+        return true;
+
+    while (*value)
+    {
+        if (!iswspace(*value))
+            return false;
+        ++value;
+    }
+    return true;
+}
+
+static inline uint32_t GetStringLength(const char *&value)
```

Comment:
Do we really need this overload? Looks like all the strings that are being written are `LPCWSTR` type so we wouldn't ever use this overload.

Reply:
I included it only because I noticed that other IPC code paths handle both `char` and `wchar`.  Despite that, I believe the IPC protocol exclusively uses `wchar`, so it is most likely not worth the code size for the `char` implementations.